instrumentation/android-instrumentation/src/main/java/io/opentelemetry/android/instrumentation/InstallationContext.kt

```diff
@@ -6,11 +6,45 @@
 package io.opentelemetry.android.instrumentation
 
 import android.app.Application
+import io.embrace.opentelemetry.kotlin.ExperimentalApi
+import io.embrace.opentelemetry.kotlin.OpenTelemetry
+import io.embrace.opentelemetry.kotlin.OpenTelemetryInstance
+import io.embrace.opentelemetry.kotlin.context.Context
+import io.embrace.opentelemetry.kotlin.createOpenTelemetryKotlin
 import io.opentelemetry.android.session.SessionProvider
-import io.opentelemetry.api.OpenTelemetry
 
+@OptIn(ExperimentalApi::class)
 data class InstallationContext(
     val application: Application,
-    val openTelemetry: OpenTelemetry,
+    val openTelemetry: io.opentelemetry.api.OpenTelemetry,
     val sessionProvider: SessionProvider,
-)
+    val openTelemetryKotlin: OpenTelemetry = OpenTelemetryInstance.createOpenTelemetryKotlin(),
+    val rootContext: Context = openTelemetryKotlin.objectCreator.context.root(),
+) {
+    override fun hashCode(): Int {
+        var result = application.hashCode()
+        result = 31 * result + openTelemetry.hashCode()
+        result = 31 * result + sessionProvider.hashCode()
+        return result
+    }
+
+    override fun equals(other: Any?): Boolean {
```

Comment:
Why are these necessary over the generated implementations the data class will provide?